### ISSUE TITLE
[Feature Request] Add support for setting IDLE in `gnus-demon-add-handler` function.

### ISSUE THREAD

**Issue by fernandodemorais-jf** [seagle0128/doom-modeline#514]
Hello!

---

**Is your feature request related to a problem? Please describe.**
There is no way to control idle timers in `gnus-demon-add-handler` called in `doom-modeline-start-gnus-listener` function. The default is always `nil`, unless we change it directly in [`doom-modeline-segments.el`](https://github.com/seagle0128/doom-modeline/blob/master/doom-modeline-segments.el) file.

This can be seen in [line #2484](https://github.com/seagle0128/doom-modeline/blob/master/doom-modeline-segments.el#L2484) of the mentioned file:

```elisp
;; ...
        (gnus-demon-add-handler 'gnus-demon-scan-news doom-modeline-gnus-timer nil))))
;; ...
```

**Describe the solution you'd like**
I believe the best way to solve this is to create a variable in [`doom-modeline-core.el`](https://github.com/seagle0128/doom-modeline/blob/master/doom-modeline-core.el), like the following (just a suggestion):

```elisp
(defcustom doom-modeline-gnus-idle nil
  "Whether to wait an idle time to scan for news.

When t, sets `doom-modeline-gnus-timer' as an idle timer.  If a
number, Emacs must have been idle this given time, checked after
reach the defined timer, to fetch news.  The time step can be
configured in `gnus-demon-timestep'."
  :type '(choice
	  (boolean :tag "Set `doom-modeline-gnus-timer' as an idle timer")
	  (number :tag "Set a custom idle timer"))
  :group 'doom-modeline)
```
And, then, pass it to `gnus-demon-add-handler`:

```elisp
;; ...
        (gnus-demon-add-handler 'gnus-demon-scan-news doom-modeline-gnus-timer doom-modeline-gnus-idle))))
;; ...
```
---

Thanks!

**Comment by seagle0128** [comment]
@Melkor333 Could you please take a look at this request? Since https://github.com/seagle0128/doom-modeline/pull/279 is related to it. Thanks!

**Comment by Melkor333** [comment]
I haven't been using `gnus` for a few years now, so I can't really comment on it that much.
But @fernandodemorais-jf your suggestion is exactly how I would've added this feature. Can you make an MR with it? :slightly_smiling_face:

**Comment by fernandodemorais-jf** [comment]
> But @fernandodemorais-jf your suggestion is exactly how I would've added this feature. Can you make an MR with it? slightly_smiling_face

Sure! :+1:

**Comment by fernandodemorais-jf** [comment]
Hello!

Closing, because this feature request was implemented in the, already merged, PR #515.

Thanks @seagle0128 and @Melkor333. :handshake:
